avro/src/test/scala/magnolify/avro/test/AvroTypeSpec.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2019 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package magnolify.avro.test
+
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream}
+
+import cats._
+import cats.instances.all._
+import magnolify.avro.{AvroType, GenericRecordType}
+import magnolify.avro.GenericRecordType._
+import magnolify.cats.auto._
+import magnolify.scalacheck.auto._
+import magnolify.test.Simple._
+import magnolify.test._
+import org.apache.avro.generic.{GenericDatumReader, GenericDatumWriter, GenericRecord}
+import org.apache.avro.io.{DecoderFactory, EncoderFactory}
+import org.scalacheck._
+
+import scala.reflect._
+
+object AvroTypeSpec extends MagnolifySpec("AvroRecordType") {
+  private val encoder = EncoderFactory.get
+  private val decoder = DecoderFactory.get
+
+  private def test[T: Arbitrary: ClassTag](
+    implicit tpe: AvroType.Aux2[T, GenericRecord],
+    eq: Eq[T]
+  ): Unit = {
+    ensureSerializable(tpe)
+    val converter = GenericRecordType[T]
+
+    property(className[T]) = Prop.forAll { caseClass: T =>
+      val avroRepr = converter.to(caseClass)
+      val avroCopy = roundtripAvro(avroRepr)
+      val copy = converter.from(avroCopy)
+
+      Prop.all(eq.eqv(caseClass, copy))
```

Comment:
I originally included a property test comparing `avroRepr` and `avroCopy` using the default GenericRecord `equals()`, and it failed on Map types when the ordering of map entries was different. a follow up project to this PR could be adding an Equality derivation for Avro types?